### PR TITLE
Rollover optimization

### PR DESCRIPTION
**Pull request by kkennis**
Do a couple small fixes to optimize the rollover contract.

* Make `executeOperation` non-reentrant
* Check 0 balances before and after flash rollover operation.
* Shave off some gas in a few places to offset the reentrancy check.

### CODE REVIEW

File: contracts/FlashRollover.sol [hunk 1]
```diff
@@ -254,24 +243,38 @@ contract FlashRollover is IFlashRollover {
...
-        require(newLoanTerms.collateralTokenId == terms.collateralTokenId, "Collateral mismatch");
         require(
-            address(loanCore.collateralToken()) == address(targetLoanCore.collateralToken()),
-            "Non-compatible AssetWrapper"
+            address(sourceLoanCore.collateralToken()) == address(targetLoanCore.collateralToken()),
+            "non-compatible AssetWrapper"
         );
     }
+
+    function setOwner(address _owner) external override {
```
Comment (marktoda) [inline contracts/FlashRollover.sol:264]: why not just use oz ownable?

**Review by marktoda — APPROVED** [review]

File: contracts/FlashRollover.sol (hunk 1 above)
Comment (kkennis) [reply contracts/FlashRollover.sol:264]: Cost 🤷  I figured the usage surface area is so small, we didn't need the full functionality of `Ownable` and could save a bit on deployment and usage - wouldn't want things like `renounceOwnership` to work in this case anyway

**Review by misamoo — COMMENTED** [review]
d

**Review by misamoo — COMMENTED** [review]
d

**Pull request merged**